Components/Sensors/BME280.cs

```diff
@@ -0,0 +1,367 @@
+            this.Humidity = (this.Configuration.MeasureHumidity ? (ProcessHumidity(RawHumidity, IntTempCal)) : double.NaN);
+        }
+
+        #region Read/Write
+        private byte[] Read(byte Register, byte Length)
+        {
+            if (this.IsSPI)
+            {
+                byte[] DataOut = new byte[Length + 1];
+                DataOut[0] = (byte)((Register & 0b0111_1111) | 0b1000_0000);
+                byte[] DataIn = this.SPIBus.Write(this.SPICS, DataOut, DataOut.Length);
+                return UtilMain.SubArray(DataIn, 1, Length);
+            }
+            else
+            {
+                return this.I2CBus.ReadRegister(this.I2CAddress, Register, Length);
+            }
+        }
+
+        /// <summary> Writes a single register. </summary>
+        /// <param name="Register"> The register address to write to. </param>
+        /// <param name="Data"> The data to write. </param>
+        private void Write(byte Register, byte Data)
+        {
+            if (this.IsSPI) { this.SPIBus.Write(this.SPICS, new byte[] { (byte)(Register & 0b0111_1111), Data }, 2); }
+            else { this.I2CBus.WriteRegister(this.I2CAddress, Register, new byte[] { Data }); }
+        }
+        #endregion
+
+        #region Data Processing
+        /// <summary> Gets temperature in format useful for compensation of other values. </summary>
+        /// <remarks> Corresponds to <c>t_fine</c> in original code. </remarks>
+        /// <source> Datasheet, page 23 </source>
+        /// <param name="RawTemp"> The temperature as read directly from the device's registers. </param>
+        private int ProcessTemperatureInternal(int RawTemp)
+        {
+            int Var1, Var2;
+            Var1 = ((((RawTemp >> 3) - (this.CompParams.T1 << 1))) * (this.CompParams.T2)) >> 11;
+            Var2 = (((((RawTemp >> 4) - (this.CompParams.T1)) * ((RawTemp >> 4) - (this.CompParams.T1))) >> 12) * (this.CompParams.T3)) >> 14;
+            return Var1 + Var2;
+        }
+
+        /// <summary> Gets the actual external temperature. </summary>
+        /// <source> Datasheet, page 23 </source>
+        /// <param name="TempInternal"> The temperature produced by <c>ProcessTemperatureInternal()</c>. </param>
+        /// <returns> The external temperature reading in increments of 0.01 degrees Celsius. </returns>
+        private double ProcessTemperature(int TempInternal)
+        {
+            return (((TempInternal * 5) + 128) >> 8) / 100.00D;
+        }
+
+        /// <summary> Gets the pressure reading. </summary>
+        /// <param name="RawPress"> The pressure as read directly from the device's registers. </param>
+        /// <param name="TempComp"> The temperature produced by <c>ProcessTemperatureInternal()</c>. </param>
+        /// <returns> Pressure reading in 1/256 increments, in Pa. </returns>
+        private double ProcessPressure(int RawPress, int TempComp)
+        {
+            long Var1, Var2, P;
```

Comment:
These sure are some interesting computations...

Reply:
They certainly are. Lots of fun to implement :)